plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageSettings.java

```diff
@@ -22,35 +22,73 @@
 import org.elasticsearch.cloud.azure.storage.AzureStorageService.Storage;
 import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.AffixSetting;
+import org.elasticsearch.common.settings.Setting.Property;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.settings.SettingsException;
 import org.elasticsearch.common.unit.TimeValue;
-import org.elasticsearch.node.Node;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.function.Function;
+
+import static org.elasticsearch.cloud.azure.storage.AzureStorageService.Storage.STORAGE_ACCOUNTS;
 
 public final class AzureStorageSettings {
-    private static final Setting<TimeValue> TIMEOUT_SETTING = Setting.affixKeySetting(Storage.PREFIX, "timeout",
-        (key) -> Setting.timeSetting(key, Storage.TIMEOUT_SETTING, Setting.Property.NodeScope));
-    private static final Setting<String> ACCOUNT_SETTING =
-        Setting.affixKeySetting(Storage.PREFIX, "account", (key) -> Setting.simpleString(key, Setting.Property.NodeScope));
-    private static final Setting<String> KEY_SETTING =
-        Setting.affixKeySetting(Storage.PREFIX, "key", (key) -> Setting.simpleString(key, Setting.Property.NodeScope));
-    private static final Setting<Boolean> DEFAULT_SETTING =
-        Setting.affixKeySetting(Storage.PREFIX, "default", (key) -> Setting.boolSetting(key, false, Setting.Property.NodeScope));
+    // prefix for azure client settings
+    private static final String PREFIX = "azure.client.";
+
+    /**
+     * Azure account name
+     */
+    public static final AffixSetting<String> ACCOUNT_SETTING = Setting.affixKeySetting(PREFIX, "account",
+        key -> Setting.simpleString(key, Property.NodeScope));
+
+    /**
+     * Azure key
+     */
+    public static final AffixSetting<String> KEY_SETTING = Setting.affixKeySetting(PREFIX, "key",
+        key -> Setting.simpleString(key, Property.NodeScope));
```

Comment:
Let's do it like this.

Can you review the first phase of this PR and then I can add more commits to implement the security part? So we merge only a good version?
Like if we were having a feature branch.

WDYT?